runtime/src/main/java/org/corfudb/runtime/object/MultiVersionObject.java

```diff
@@ -327,6 +327,10 @@ private void syncStreamUnsafe(long timestamp) {
                         addressSpace.addAddress(globalAddress);
                         materializedUpTo = globalAddress;
                         resolvedUpTo = globalAddress;
+                    } catch (IllegalStateException e) {
+                        // The caller catches this TrimmedException and resets the object before retrying.
+                        log.error("Sync[{}] couldn't sync to {} due to {}", Utils.toReadableId(getID()), timestamp, e);
+                        throw new TrimmedException();
                     } catch (Exception e) {
```

Comment:
Just wondering if we still need this case - at what Exception will we need to throw an UnrecoverableCorfuError which clearly makes the state inconsistent?
Also the caller (MultiVersionObject.java) already seems to incorporate the trim retry logic where it resets the state, can it not be the default failure behavior?

Reply:
Good question. I kept this case because an exception can be thrown by `applyUpdateUnsafe` if the SMR update at that address contains an SMR method that is not part of this object's `upcallTargetMap`. In that case, the MVO does not know how to update its in-memory view of the object, and is an indication that something else has gone awry. Retrying here will not help and I don't believe that propagating the TrimmedException to the client is appropriate either.